### PR TITLE
Apriltag localization

### PR DESCRIPTION
**Pull request by gaborszita**

### CODE REVIEW

File: src/main/cpp/AprilTags.cpp [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+#include <units/length.h>
+#include <frc/DriverStation.h>
+#include "Constants.h"
+
+
+using namespace units::length;
+using namespace frc;
+
+frc::AprilTagFieldLayout AprilTags::getAprilTagFieldLayout() {
+    // if blue alliance or invalid
```
Comment (SusanC3) [inline src/main/cpp/AprilTags.cpp:12]: Is it necessary to have different cases for being on the blue vs red alliance? isn't the field symmetrical?

**Review by NGC3131224 — COMMENTED** [review]
I doubled checked the measurements and found a few that seemed off from given values. New values were given in field diagrams (+calculations) or cad measurements.

+- [error value in inches] for [measurement name] ([current value in code]-[new value]) ; [other differing calcs]

-3.14 error for apriltag_x (37-40.14)	; 40.45 given by LayoutMarkingDiagram table; 40.14 from cad
-0.25 error for substation_apriltag_x (14-14.25)
-1.4275 error for substation_apriltag_y (264.3125-265.74)	;calc from LayoutMarkingDiagram dimensions indicate 265.675; 265.46 from cad

Table of given values:
Screenshot 2023-01-29 155646

Layout Marking Diagram Link: https://firstfrc.blob.core.windows.net/frc2023/FieldAssets/2023LayoutMarkingDiagram.pdf
Cad model Link: https://firstfrc.blob.core.windows.net/frc2023/FieldAssets/CHARGEDUP2023Field-STEP.zip

**Comment by gaborszita** [comment]
I didn't know this table existed - at the time I did this, I was using the game manual to calculate the apriltag positions. It required me to add and subtract a lot of distance values on the field, so accuracy was probably lost because values on the game manual aren't millimeter-precise. I'll update the values as those in the table.

**Comment by gaborszita** [comment]
FIXED

@NGC3131224

**Comment by linc-mvla** [comment]
Desmos tests math, successful: https://www.desmos.com/calculator/gvkoxf1rrf

**Review by linc-mvla — APPROVED** [review]

File: src/main/cpp/AprilTags.cpp (hunk 1 above)
Comment (SusanC3) [reply src/main/cpp/AprilTags.cpp:12]: Field has mirror symmetry not rotational symmetry so it doesn't work (screw FRC)

**Review by SusanC3 — APPROVED** [review]

**Pull request merged**